### PR TITLE
[all] Homogenize api update

### PR DESCRIPTION
**Pull request by hugtalbot**
Homogeneize the API for update.
A layer is added to avoid a direct access to the DataTracjer which is now private:
- in DataTrackerDDGNode: trackInputData, hasInputDataChanged, haveInputsDataChanged, cleanInputTracker
- in BaseObject: trackData, hasDataChanged

The name of the update function is:
- `doUpdate()` for Engines
- `doInternalUpdate()` for components inheriting from BaseObject

Update engines, and tests accordingly

Rename UpdateInternal(DataVisitor) into InternalUpdate(DataVisitor)

Remove the DataTrackerFunctor and its test

TODO: update constantFF with new API

______________________________________________________

This PR: 
- [ ] builds with SUCCESS for all platforms on the CI.
- [ ] does not generate new warnings.
- [ ] does not generate new unit test failures.
- [ ] does not generate new scene test failures.
- [ ] does not break API compatibility.
- [ ] is more than 1 week old (or has fast-merge label).

**Reviewers will merge only if all these checks are true.**

### CODE REVIEW

File: SofaKernel/framework/sofa/core/objectmodel/BaseObject.cpp [hunk 1]
```diff
@@ -274,30 +274,30 @@ void BaseObject::reinit()
 {
 }
 
-void BaseObject::updateInternal()
+void BaseObject::internalUpdate()
```
Comment (damienmarchal) [inline SofaKernel/framework/sofa/core/objectmodel/BaseObject.cpp:277]: Wow we are now accepting super breaking renaming :)

File: SofaKernel/modules/SofaDeformable/StiffSpringForceField.inl [hunk 2]
```diff
@@ -60,8 +60,8 @@ void StiffSpringForceField<DataTypes>::init()
 {
     if (d_indices1.isSet() && d_indices2.isSet())
     {        
-        this->trackInternalData(d_indices1);
-        this->trackInternalData(d_indices2);
+        this->trackData(d_indices1);
```
Comment (damienmarchal) [inline SofaKernel/modules/SofaDeformable/StiffSpringForceField.inl:63]: rename trackInternalData

File: SofaKernel/framework/sofa/core/DataTracker.h [hunk 3]
```diff
@@ -96,28 +96,39 @@ namespace core
...
+        void trackInputData(const objectmodel::BaseData &data);
+
+        /// Function checking if a specific input data changed
+        bool hasInputDataChanged(const objectmodel::BaseData &data);
+
+        /// Function checking if one or several inputs did changed
+        bool haveInputsDataChanged();
 
+        /// Function cleaning the DataTracker for inputs
+        void cleanInputTracker();
```
Comment (damienmarchal) [inline SofaKernel/framework/sofa/core/DataTracker.h:115]: rename to cleanInputDataTracker()

File: modules/SofaMiscForceField/MeshMatrixMass.inl [hunk 4]
```diff
@@ -929,10 +929,10 @@ void MeshMatrixMass<DataTypes, MassType>::init()
     f_graph.endEdit();
 
     // add data to tracker
-    this->trackInternalData(d_vertexMass);
-    this->trackInternalData(d_edgeMass);
-    this->trackInternalData(d_massDensity);
-    this->trackInternalData(d_totalMass);
+    this->trackData(d_vertexMass);
```
Comment (damienmarchal) [inline modules/SofaMiscForceField/MeshMatrixMass.inl:932]: trackInternalData :)

File: modules/SofaMiscForceField/MeshMatrixMass.inl [hunk 5]
```diff
@@ -1290,13 +1290,13 @@ void MeshMatrixMass<DataTypes, MassType>::computeMass()
...
+    // Now update is handled through the doInternalUpdate mechanism
+    // called at each begin of step through the InternalUpdateDataVisitor
 }
 
 
 template <class DataTypes, class MassType>
-void MeshMatrixMass<DataTypes, MassType>::doUpdateInternal()
+void MeshMatrixMass<DataTypes, MassType>::doInternalUpdate()
 {
     if (this->hasDataChanged(d_totalMass))
```
Comment (damienmarchal) [inline modules/SofaMiscForceField/MeshMatrixMass.inl:1301]: hasInternalDataChanged

**Comment by guparan** [comment]
I close this for now. It has diverged a lot and I'm not even sure it is applicable to the current SOFA.

**Pull request closed without merging**